src/relay/backend/contrib/ethosn/ethosn_api.cc

```diff
@@ -171,6 +171,19 @@ EthosnError EthosnAPI::QnnConv2d(const Expr& expr, ConvolutionParams* params) {
   return err;
 }
 
+Constant TransposeWeights(const Constant& data, const std::string& input_layout,
```

Comment:
Just a thought: Does it make sense to have it common for contrib or available for other passes?
One other use case: https://github.com/apache/tvm/blob/fa17da22c73fb9e95c27e4c28130835b628caf6b/src/relay/backend/contrib/cmsisnn/generate_constants.cc#L74